packages/aws-cdk/lib/api/aws-auth/account-cache.ts

```diff
@@ -77,18 +77,25 @@ export class AccountAccessKeyCache {
   }
 
   private async loadMap(): Promise<{ [accessKeyId: string]: Account }> {
-    if (!(await fs.pathExists(this.cacheFile))) {
-      return { };
+    try {
+      return await fs.readJson(this.cacheFile);
+    } catch (e) {
+      if (e.code === 'ENOENT' || e.code === 'EACCES') { return {}; }
+      throw e;
     }
-
-    return await fs.readJson(this.cacheFile);
   }
 
   private async saveMap(map: { [accessKeyId: string]: Account }) {
-    if (!(await fs.pathExists(this.cacheFile))) {
-      await fs.mkdirs(path.dirname(this.cacheFile));
-    }
+    try {
+      const dir = path.dirname(this.cacheFile);
+      if (!(await fs.pathExists(dir))) {
+        await fs.mkdirs(dir);
+      }
```

Comment:
why not something like `fs.ensureFileSync(this.cacheFile);` which will create the directories if it doesn't exist

Reply:
Sure.